File: backend/time/models/syncfusion.go
package models

import (
	"time"
)

type SyncfusionGantt struct {
	Id        uint32
	Name      string
	Relate    string
	RelateId  int
	Parent    uint32
	Progress  int
	Duration  int
	Expanded  bool
	Status    int
	Color     string
	StartDate time.Time
	EndDate   time.Time
	Children  []SyncfusionGantt
}

var (
	SyncfusionGanttLevelQuest   = "quest"
	SyncfusionGanttLevelProject = "project"
	SyncfusionGanttLevelTask    = "task"
	SyncfusionGanttStatusWait   = "wait"
	SyncfusionGanttStatusFinish = "finish"
)

type SyncfusionTreeGrid struct {
	Id         int
	Name       string
	Parent     interface{}
	IsParent   bool
	IsExpanded bool
	Children   []SyncfusionTreeGrid `json:"Children,omitempty"`
}

type SyncfusionSchedule struct {
	Id         int
	Name       string
	StartDate  time.Time
	EndDate    time.Time
	FieldId    string
	AllDay     bool
	Recurrence bool
	Area       Area
	Task       Task
	Resource   Resource
	Field      Field
}

type SyncfusionKanban struct {
	Id           int
	Name         string
	Desc         string
	Tags         string
	Status       int
	StatusName   string
	ResourceId   int
	ResourceName string
	ProjectId    int
	ProjectName  string
	FieldId      int
	FieldName    string
}

var (
	SyncfusionScheduleViewWeek     string = "week"
	SyncfusionScheduleViewDay      string = "day"
	SyncfusionScheduleViewMonth    string = "month"
	SyncfusionScheduleViewAgenda   string = "agenda"
	SyncfusionScheduleViewWorkWeek string = "workweek"
)

File: frontend/src/app/model/time/syncfusion.ts
import { Basic }  from '../base/basic';

export class TreeGrid extends Basic {
  Id: number;
  Name: string;
  Parent: string;
  ParentId: number;
  IsParent: boolean;
  Children: TreeGrid[];
  constructor (json?: any) {
    if (json != undefined) {
      super(json)
    } else {
      this.Children = [];
    }
  }
}

export class Gantt extends Basic {
  Id: number;
  Name: string;
  Relate: string;
  StartDate: Date;
  EndDate: Date;
  Progress: number;
  Duration: number;
  Expended: boolean;
  Parent: number;
  Status: number;
  Color: string;
  Children: Gantt[];
  constructor (json?: any) {
    if (json != undefined) {
      super(json)
    } else {
      this.Children = [];
    }
  }
}

export class Schedule extends Basic {
  Id: number;
  Name: string;
  StartDate: Date;
  EndDate: Date;
  AllDay: boolean;
  Recurrence: boolean;
}

export class Kanban extends Basic {
	Id          :number
	Name        :string
	Desc        :string
	Status      :number
	StatusName  :string
	ProjectName :number
	Tags        :string
	FieldName   :string
	ResourceId  :number
	ProjectId   :number
	FieldId     :number
}
